### PR TITLE
Do not set the adapter on faraday prior the request

### LINKED ISSUE #148: WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
This seems to get cause by diplomat nowdas:
```
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
```

see https://github.com/lostisland/faraday/commit/067be864a67e791e6766253ce4ef3f5c740caf70

### PR DESCRIPTION
**Pull request by EugenMayer**
Related to https://github.com/lostisland/faraday/pull/685 and fixes #148

### CODE REVIEW

**Comment by EugenMayer** [comment]
I case this is kind of counting as "cosmetic" - it is not really.

Any logs are useless, since every single consul request logs this line and you basically flood your logs, your servers and this becomes completely nonsense:

```
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
WARNING: Unexpected middleware set after the adapter. This won't be supported from Faraday 1.0.
```
This continues all day long, gather several 100mb of logs. Please fix this ASAP, i beg you, or at least downgrade faraday to the version you had before ( 0.8.x)

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (johnhamelink) [inline .idea/vcs.xml]: @EugenMayer Could you please remove the `.idea` directory from the PR and add `.idea` to the `.gitignore`?

**Comment by EugenMayer** [comment]
@johnhamelink sorry - not clear how i did add this one in .. not intended - removed

**Comment by EugenMayer** [comment]
any ETA when this gets released? i had quiet an effort to move all my stacks to my forked development version .. pitty.

**Comment by johnhamelink** [comment]
@EugenMayer I just released it as version 2.0.0

**Comment by EugenMayer** [comment]
@johnhamelink thanks - major release. Any breaking changes? Sounds like nothing i want to switch to, i rather was hoping for 1.3.1 actually - no major stuff?

Could you explain a bit what 2.0.0 does include?

**Comment by johnhamelink** [comment]
@EugenMayer the 2.0.0 release includes this PR: https://github.com/WeAreFarmGeek/diplomat/pull/147

**Comment by EugenMayer** [comment]
oh great, a long time needed fix. Great!

**Comment by johnhamelink** [comment]
No problem @EugenMayer :) Thanks for using Diplomat!

**Pull request merged**